src/backend/distributed/shardsplit/cdc_decoder.c

```diff
@@ -0,0 +1,388 @@
+/*-------------------------------------------------------------------------
+ *
+ * cdc_decoder.c
+ *		CDC Decoder plugin for Citus
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+#include "common/hashfn.h"
+#include "utils/typcache.h"
+#include "utils/lsyscache.h"
+#include "catalog/pg_namespace.h"
+#include "distributed/cdc_decoder.h"
+#include "distributed/relay_utility.h"
+#include "distributed/worker_protocol.h"
+#include "distributed/metadata_cache.h"
+
+static LogicalDecodeChangeCB ouputPluginChangeCB;
+
+
+static bool replication_origin_filter_cb(LogicalDecodingContext *ctx, RepOriginId
+										 origin_id);
+
+static void TranslateChangesIfSchemaChanged(Relation relation, Relation targetRelation,
+											ReorderBufferChange *change);
+
+static void TranslateAndPublishRelationForCDC(LogicalDecodingContext *ctx,
+											  ReorderBufferTXN *txn,
+											  Relation relation,
+											  ReorderBufferChange *change, Oid shardId,
+											  Oid targetRelationid);
+
+typedef struct
+{
+	Oid shardId;
```

Comment:
should be uint64?

Reply:
Sure I will change it to uint64.
But since this is a key in hash table, we need the "hash" function.
Currently oid_hash is the hash function, but I don't find any "uint64_hash" function..
Can I still use oid_hash function after changing it to uin64?